app/controllers/assignments_controller.rb

```diff
@@ -29,7 +29,8 @@ def edit; end
 
   def update
     if @assignment.update_attributes(update_assignment_params)
-      flash[:success] = "Assignment \"#{@assignment.title}\" updated"
+      AssignmentVisibilityJob.perform_later(@assignment) if @assignment.previous_changes['public_repo']
```

Comment:
I'm using `previous_changes` here, but I'm not sure if this is a preferable way to detect changes.